tests/TestCase.php

```diff
@@ -2,9 +2,26 @@
 
 namespace Tests;
 
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function test_hasMany_relation($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function test_belongsTo_relation($related, $foreignKey, $ownerKey, $relation)
+    {
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($ownerKey, $relation->getOwnerKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
```

Comment:
em test những gì ở đây?

Reply:
$this->assertInstanceOf(HasMany::class, $relation): kiểm tra xem có phải instance của HasMany?
$this->assertInstanceOf($related, $relation->getRelated()): kiểm tra kiểu dữ liệu
$this->assertEquals($foreignKey, $relation->getForeignKeyName()): kiểm tra khóa ngoại.
$this->assertInstanceOf(BelongsTo::class, $relation): kiểm tra xem có phải instance của BelongsTo
$this->assertEquals($ownerKey, $relation->getOwnerKeyName()): nhận về tên của khóa chính
 $this->assertEquals($foreignKey, $relation->getForeignKeyName()): test foreignkey